datavec/datavec-data/datavec-data-image/src/main/java/org/datavec/image/loader/NativeImageLoader.java

```diff
@@ -824,36 +838,37 @@ private INDArray asMatrix(BytePointer bytes, long length) throws IOException {
         INDArrayIndex[] index = null;
         switch (this.multiPageMode) {
             case MINIBATCH:
-                data = Nd4j.create(pixa.n(), 1, pixa.pix(0).h(), pixa.pix(0).w());
+                data = Nd4j.create(pixa.n(), 1, 1, pixa.pix(0).h(), pixa.pix(0).w());
```

Comment:
We want to set the depth and the number of channels here instead of just "1, 1"

Reply:
Did I miss somethings ? What do you mean depth here ? Depth of image of depth of 3d volume ?

Reply:
I believe you mean depth of 3D volume. But in that case, I can not guess the structure of data. I need more metadata information to be able to correctly load the image. For example, what are the numbers of channel/depth/batch?  How are they arranged ? If you need data at this complexity. I recommend you to wrap this project [tifffile](https://pypi.org/project/tifffile/) into Datavec. I saw you began to work with numpy. It won't be much difficult, if you are already able to communicate numpy with INDArray

Reply:
Or this Java project ([bioformat](https://www.openmicroscopy.org/bio-formats/)), if you don't want Python.

Reply:
It's not about NDArray, it's about returning the data that is in a TIFF file, but you're right, `pixa.n()` is often basically the "depth",  so we'd need another DEPTH mode would be a good name I think:
```
Nd4j.create(1, channels, pixa.n(), pixa.pix(0).h(), pixa.pix(0).w());
```
But sometimes we want to return it as a batch though, so for MINIBATCH mode we'd have:
```
Nd4j.create(pixa.n(), channels, 1, pixa.pix(0).h(), pixa.pix(0).w());
```
For the CHANNELS mode, I guess something like this:
```
Nd4j.create(1, pixa.n(), 1, pixa.pix(0).h(), pixa.pix(0).w());
```
But the channels information goes away and this is strange... What about supporting only the first one above, the one in NCDHW order, and leave it up to the user to manipulate the output further if they need it?

Reply:
Sure, we can use other dependencies than Leptonica if something else is easier to work with.

Reply:
Sincerely, I personally don't think hard-coding this part is a good idea. It seems that you started to have biomedical image users. In this area, your 5 channels is still not enough. You will need the time dimension. Generally, we (at least me) work with by default 5-channel images (TimeChannelZXY). For the training, you need anther N. That become really complex from my point of view. What do you think @AlexDBlack ?

Reply:
@saudet I believe for instance its better to let user permute their data axis to get what they want, because integrating metadata to get exactly what they want is **painful**...

Reply:
I don't think TIFF files contain other axis like time, right? If so let's not worry about that. We're only interested in loading data image files here.

Reply:
It depends on what tiff you want to manipulate. 

So, yes. In this case, In fact, I would remove the multipage option. Every image will by default be loaded in `batch` channel. It up to user to reorder the image. Is this what you wanted ?

Reply:
Sounds reasonable, yes. Let's just make sure images with more than 1 channel can be loaded.

Reply:
This still doesn't take into account the number of channels in the image. We need 
to put the `channels` variable somewhere here and below as well.